src/helpers/logging-interceptor.helper.ts

```diff
@@ -0,0 +1,19 @@
+import { CallHandler, ExecutionContext, Injectable, Logger, NestInterceptor } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
+export class LoggingInterceptor implements NestInterceptor {
+  private readonly logger = new Logger();
```

Comment:
All our other classes that use a logger inject it through the constructor - is that not an option here?

Reply:
As discussed yesterday, the NestJS Logger object doesn't seem to support dependency injection, whereas the Pino Logger does. Using the Pino Logger instead isn't an issue as the NestJS Logger and Pino Logger are one and the same anyway (due to us overriding the default NestJS Logger globally), it just means we'd have to change the logger in several places if we ever decided to change logger. We decided this sacrifice was worth it in order to be able to use dependency injection, especially if we ever write tests for this logging (I've raised a ticket for this https://ukef-dtfs.atlassian.net/browse/APIM-318)